### PR TITLE
Preparing for 4.5.1 on rocky/alma 9

### PR DESCRIPTION
**Pull request by sandrobonazzola**
## Changes introduced with this PR

* backport to 4.5 branch for PR #50

## Are you the owner of the code you are sending in, or do you have permission of the owner?

y

### CODE REVIEW

**Comment by sandrobonazzola** [comment]
this is currently failing on 

12:35:04,287 INFO anaconda:program: Problem: package ovirt-release-host-node-4.5.0.3-1.el9.x86_64 requires ovirt-host, but none of the providers can be installed
12:35:04,287 INFO anaconda:program: - cannot install the best candidate for the job
12:35:04,287 INFO anaconda:program: - nothing provides scap-security-guide >= 0.1.60-4 needed by ovirt-host-4.5.0-3.el9.x86_64

File: build.sh [hunk 1]
```diff
@@ -47,23 +47,40 @@ build() {
...
-    esac
-
+    if [ "$(rpm --eval %{almalinux})" == "9" ] ; then
+                prepare_osinfo_db
+                # Replace SSG TARGET with ssg-rhel9-ds.xml equivalent
+                export SSG_TARGET_XML=/usr/share/xml/scap/ssg/content/ssg-rhel9-ds.xml
+                export SHIP_OVIRT_CONF=1
+                ./autogen.sh \
+                    --with-distro=alma9 \
+                    --with-bootisourl="https://repo.almalinux.org/almalinux/9/isos/x86_64/AlmaLinux-9.0-beta-1-x86_64-boot.iso"
```
Comment (srbala) [inline build.sh:57]: Almalinux 9 GA is available now, please update the URL.
```
--with-bootisourl="http://repo.almalinux.org/almalinux/9/isos/x86_64/AlmaLinux-9-latest-x86_64-boot.iso"
```

**Review by srbala — COMMENTED** [review]
Update the URL, we are good to go.

**Comment by srbala** [comment]
> Alma has: scap-security-guide.noarch 0.1.57-5.el9
> 
> RHEL 9 has scap-security-guide-0.1.60-6.el9_0 so it should be ok by Alma 9 GA

AlmaLinux 9, GA is available with an updated package, the version looks good.

File: build.sh [hunk 2]
```diff
@@ -47,23 +47,40 @@ build() {
...
-            ./autogen.sh \
-                --with-distro=c9s \
-                --with-bootisourl=http://mirror.stream.centos.org/9-stream/BaseOS/x86_64/iso/CentOS-Stream-9-latest-x86_64-boot.iso
-            ;;
-    esac
-
+    if [ "$(rpm --eval %{almalinux})" == "9" ] ; then
+                prepare_osinfo_db
+                # Replace SSG TARGET with ssg-rhel9-ds.xml equivalent
+                export SSG_TARGET_XML=/usr/share/xml/scap/ssg/content/ssg-rhel9-ds.xml
```
Comment (srbala) [inline build.sh:53]: Please update `ssg-rhel9-ds.xml` to `ssg-almalinux9-ds.xml`

File: data/distro-defs.yml [hunk 3]
```diff
@@ -60,3 +58,69 @@ c9s:
...
+        dnf distro-sync --nobest
+        dnf install -y centos-release-ovirt45
+  packages-switch: --excludeWeakdeps
+  packages:
+    - dracut-live
+    - centos-stream-repos
+    - scap-security-guide  # needed for post action
+  post:
+    - |
+        ln -sf /usr/share/xml/scap/ssg/content/{ssg-rhel9,ssg-onn45}-ds.xml
```
Comment (srbala) [inline data/distro-defs.yml:93]: And here `ssg-rhel9` to `ssg-almalinux`

Comment (sandrobonazzola) [reply data/distro-defs.yml:93]: thanks

File: build.sh (hunk 2 above)
Comment (sandrobonazzola) [reply build.sh:53]: thanks

**Review by srbala — APPROVED** [review]

**Comment by sandrobonazzola** [comment]
Currently failing on AlmaLinux 9.0 with:
`nothing provides nmstate-plugin-ovsdb needed by vdsm-network-4.50.0.13-1.el9.x86_64` and it seems that subpackage is missing in Alma while it's available in CentOS Stream https://kojihub.stream.centos.org/koji/buildinfo?buildID=17369 @srbala can you please check?

**Comment by sandrobonazzola** [comment]
Almalinux 9.0 based node built manually and uploaded for testing at https://resources.ovirt.org/repos/testing/alma/ ; still waiting for Rocky Linux 9 for testing that build too. Merging this in the meanwhile.

**Pull request merged**